### DISCUSSION TITLE
ADR DS Architecture

### DISCUSSION THREAD

**Discussion by andresin87** [SUI-Components/sui-components#2461] [Ideas and new features]
The main goal about this is to elaborate an strategy to tackle the transition of our DS from *SCSS* to **native css** using *PostCSS*

#  Design System Architecture Decision Record

## Status

- [x] 🚧 begetting
- [ ] ❔proposed
- [ ] ✅ accepted
- [ ] ❌ rejected
- [ ] ☠ deprecated
- [ ] ↪️ superseded

## Context

The current status of our SCSS architecture is not providing us a good developer experience due to many reasons:
- Slow CI workflow steps for style files of our sites.
- Slow Build process local
- Many loops generates huge css.
- @kikoruiz @rmoralp @arnau-rius @andresz1 can you please edit that and add many more?

## Decision

The curren

**Comment by andresin87** [comment]
# Other DS architectures
- [Material UI – mui](https://github.com/SUI-Components/sui-components/discussions/2461#discussioncomment-4105496)
- [Chakra UI](https://github.com/SUI-Components/sui-components/discussions/2461#discussioncomment-4105498)
- [Mantine UI](https://github.com/SUI-Components/sui-components/discussions/2461#discussioncomment-4105501)
- [AntDesign](https://github.com/SUI-Components/sui-components/discussions/2461#discussioncomment-4105505)
- [Radix UI](https://github.com/SUI-Components/sui-components/discussions/2461#discussioncomment-4105508)

**Reply by andresin87** [reply]
## Material UI
 
They have the entire solution under js with two packages:
- [styles](https://github.com/mui/material-ui/tree/f002835bcf8cdcc9c95e1144487fcd09ba892110/packages/mui-material/src/styles)
- [colors](https://github.com/mui/material-ui/tree/f002835bcf8cdcc9c95e1144487fcd09ba892110/packages/mui-material/src/colors)

For customizing they have
- [customization](https://github.com/mui/material-ui/tree/27f1c282cf4a8c5b5aa175b8b594d3a726a8474b/docs/data/material/customization/color)

It is using css in js via [`@emotion/styled`](https://emotion.sh/docs/styled) [ref](https://github.com/mui/material-ui/blob/f002835bcf8cdcc9c95e1144487fcd09ba892110/packages/mui-material/package.json#L45). All its colors are provioded by a ThemeProvider on wich you can define your custom color palette or extend the existant. [ref](https://github.com/mui/material-ui/blob/27f1c282cf4a8c5b5aa175b8b594d3a726a8474b/docs/data/material/customization/color/color.md)

[Primer](https://primer.style/react/theming) (Github DS) is using a similar approach using [styled-system/css](https://styled-system.com/) wich uses css-in-js
___
#### [website](https://mui.com/)  [repo](https://github.com/mui/material-ui)

**Reply by andresin87** [reply]
## Chackra UI
its proposal uses
- [@chakra-ui/styled-system](https://github.com/chakra-ui/chakra-ui/tree/903aa1418d1dcac1f46e1db56a96e01a1b525f8d/packages/components/styled-system)
- [@chakra-ui/system](https://github.com/chakra-ui/chakra-ui/tree/78b729ec87cad352a85db077725d058d75f554de/packages/components/system)
this uses styled components `@emotion/styled`

**Reply by andresin87** [reply]
## Mantine UI

**Reply by andresin87** [reply]
## AntDesign

**Reply by andresin87** [reply]
## Radix UI

**Reply by arnau-rius** [reply]
Maybe we should focus first in DS that use pure CSS solutions (or PostCSS)?
